### PR TITLE
fix: Broken published package

### PR DESCRIPTION
**Pull request by JakeGinnivan**
Broken due to paths in package.json. Also upgraded a few dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
     "name": "typescript-object-validator",
     "version": "0.0.0",
     "description": "TypeScript first object validator",
-    "main": "dist/index.js",
-    "types": "dist/index.d.ts",
+    "main": "dist/cjs/index.js",
```
Comment (JakeGinnivan) [inline package.json:5]: Here is the issue I'm fixing

File: src/index.ts [hunk 2]
```diff
@@ -30,17 +30,17 @@ export type ValidatedType<
...
     : T extends 'boolean'
     ? boolean
-    : T extends 'string'[]
+    : T extends Array<'string'>
     ? string[]
-    : T extends 'number'[]
+    : T extends Array<'number'>
     ? number[]
-    : T extends 'boolean'[]
+    : T extends Array<'boolean'>
```
Comment (andyjdavis) [inline src/index.ts:37]: Is this a desirable change?

Comment (JakeGinnivan) [reply src/index.ts:37]: Yeah, it is bringing it inline with our other repos. A string literal is treated as a complex type. `boolean[]` would be fine, but `'boolean'[]` is not

**Review by andyjdavis — APPROVED** [review]

**Pull request merged**